src/main/java/sdmExam/Board.java

```diff
@@ -80,4 +87,50 @@ protected Stone colorWithCompleteChain() {
     protected Stream<Intersection> getEmptyIntersections() {
         return intersections.stream().filter(intersection -> !intersection.isOccupied());
     }
+
+    //TODO: don't know if it's useful to get the territories or just act on them
+    public List<Set<Intersection>> getTerritories() {
+        return regionsContainer.getRegions().stream()
+                .filter(region -> region.stream().allMatch(this::isOrthogonalToAtLeastTwoStones))
+                .collect(Collectors.toList());
+    }
+
+    private boolean isOrthogonalToAtLeastTwoStones(Intersection intersection) {
+        return intersections.stream()
+                .filter(intersection::isOrthogonalTo)
+                .filter(Intersection::isOccupied)
+                .count() >= 2;
+    }
+
+    public List<Intersection> getOrthogonalAdjacencyIntersections(Intersection intersection) {
+        return intersections.stream()
+                .filter(otherIntersection -> otherIntersection.isOrthogonalTo(intersection))
+                .collect(Collectors.toList());
+    }
+
+    public void fillTerritory(List<Intersection> territory, Player lastPlay){
+        List<Intersection> intersectionsAdjToTerritory;
+
+        intersectionsAdjToTerritory = territory.stream().
+                flatMap(i -> getOrthogonalAdjacencyIntersections(i).stream()).
+                filter(Intersection::isOccupied).
+                distinct().collect(Collectors.toList());
+
+        int countOfWhiteStones = (int) intersectionsAdjToTerritory.stream().
+                filter(i -> i.getStone().equals(Stone.WHITE)).count();
+        int countOfBlackStones = (int) intersectionsAdjToTerritory.stream().
+                filter(i -> i.getStone().equals(Stone.BLACK)).count();
+
+        Stone territoryStoneColor;
+
+        if(countOfWhiteStones != countOfBlackStones) {
+            territoryStoneColor = (countOfWhiteStones < countOfBlackStones) ? Stone.BLACK : Stone.WHITE;
+        }else{
+            territoryStoneColor = lastPlay.getColor().getOppositeColor();
+        }
+
+        territory.forEach(i -> i.setStone(territoryStoneColor));
```

Comment:
by using directly Intersection::setStone, we do not update the chains and the region graphs, you must use method Board:.addStoneAt

Reply:
done